spec/unit/models/project_creator_spec.rb

```diff
@@ -71,4 +71,19 @@
       GoodData::Model::ProjectCreator.pick_correct_chunks(@data, update_preference: { cascade_drops: true, preserve_data: false, unmeetable_condition: true })
     end.to raise_error
   end
+
+  it 'should pick correct update chunks based on new parameter "allow_cascade_drops" in your preference' do
+    chunk = GoodData::Model::ProjectCreator.pick_correct_chunks(@data, update_preference: { allow_cascade_drops: false })
+    expect(chunk).to eq [@chunk_a]
+  end
+
+  it 'should pick correct update chunks based on new parameter "keep_data" in your preference' do
+    chunk = GoodData::Model::ProjectCreator.pick_correct_chunks(@data, update_preference: { keep_data: false })
+    expect(chunk).to eq [@chunk_a, @chunk_b]
+  end
+
+  it 'should use new parameters in your preference instead of the old ones' do
+    chunk = GoodData::Model::ProjectCreator.pick_correct_chunks(@data, update_preference: { allow_cascade_drops: true, cascade_drops: false })
```

Comment:
`allow_cascade_drops: true` contradicts  `cascade_drops: false`. IMHO, mixing the original parameters (cascade_drops, preserve_data) with new ones (allow_ cascade_drops, keep_data) should result in an error.